sources/packages/backend/apps/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -212,6 +235,47 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
     return found.length === locations.length;
   }
 
+  /**
+   * Check if location code is already registered for the institution.
+   * @param locationCode location code.
+   * @param locationId location id in case it is an update of the location.
+   * @param institutionId institution id in case it is a new location.
+   * @returns true in case there is already a location code for the institution.
+   */
+  async hasLocationCodeForInstitution(
+    locationCode: string,
+    locationId?: number,
+    institutionId?: number,
+  ): Promise<boolean> {
+    const institutionIdQueryBuilder = await this.repo
+      .createQueryBuilder("sublocation")
+      .select("sublocation.institution.id")
+      .where("sublocation.id = :locationId", { locationId });
+
+    const queryBuilder = this.repo
+      .createQueryBuilder("location")
+      .select("1")
+      .where("location.institutionCode = :locationCode", { locationCode });
+
+    if (locationId) {
+      queryBuilder
```

Comment:
Please create the sub-query inside the `if` since it is not needed in the outside scope.

Reply:
ok! Done. Thanks!